src/reducers/updateThoughts.ts

```diff
@@ -104,16 +104,20 @@ const updateThoughts = (state: State, { thoughtIndexUpdates, contextIndexUpdates
     }),
 
     // Reset cursor on first load. The pullQueue can determine which contexts to load from the url, but cannot determine the full cursor (with ranks) until the thoughts have been loaded. To make it source agnostic, we decode the url here.
-    // state.cursor should always be null here, but check to make sure we're not overriding it
-    !state.cursorInitialized && !state.cursor
+    !state.cursorInitialized
       ? state => {
         const { contextViews, path } = decodeThoughtsUrl(state, window.location.pathname)
         const cursorNew = !path || isRoot(path) ? null : path
-        return {
+        const isCursorLoaded = cursorNew?.every(child => getThought(state, child.value))
```

Comment:
Note the use of the `getThought` selector (which should really be called `getLexeme`). You'll get to know the different selectors and util functions over time. `thoughtIndex` and `contextIndex` are low level so you will usually never access them directly.

Reply:
Also regarding terminology: a [`Path`](https://github.com/cybersemics/em/wiki/Technical-Overview#314-path) consists of `Child` nodes from different `Parents`. So I renamed `pathItem` to `child`.

Reply:
👍🏻